src/sympc/protocol/aby3/aby3.py

```diff
@@ -109,15 +123,36 @@ def truncate(x: MPCTensor, session: Session) -> MPCTensor:
         # RSPointer - public ops, Tensor Pointer - Private ops
         ptr_list = []
         ptr_name = x.share_ptrs[0].__name__
+
+        # TODO:Shoud be concised,lot of branching done,to improve communication efficiency.
+
         if ptr_name == "ReplicatedSharedTensorPointer":
-            ptr_list.append(x.share_ptrs[0].get_shares()[0].get_copy())
-            ptr_list.extend(x.share_ptrs[1].get_copy().shares)
+
+            if ring_size in {2, PRIME_NUMBER}:
+                share_ptrs = x.share_ptrs
```

Comment:
Q: For 2 we do not need truncation right? but why don't we need for `PRIME_NUMBER`?

Reply:
Good Ques, binary values we do not need encoding,as the input is only an integral value{0,1}. For Prime order rings,we use it to create mainly secret share of bits {0,1} ,so we do not need encoding.Even when do bit injection of binary bit to general ring size(`2**64,2***32`) we do not need encoding as it is sharing of bits, so we set the (`encoder_precision=0`) .This idea is the crux of the PR, where we take config info from the underlying tensors to automatically prevent truncation.

Reply:
NOTE: We take the info to truncate or not ,only from the first operand.I also use this to prevent re-encoding in select shares.
https://github.com/OpenMined/SyMPC/blob/45274eeb98f297dac04067226b2b801a1072a58a/src/sympc/protocol/falcon/falcon.py#L439

Reply:
I will ping you tomorrow about this. Wouldn't ring "2" offer he possibility to have shares of bits?

Reply:
Sure George :+1: